suite-native/module-connect-device/src/components/PinFormControlButtons.tsx

```diff
@@ -27,23 +32,47 @@ type NavigationProp = StackToStackCompositeNavigationProps<
 >;
 
 export const PinFormControlButtons = () => {
+    const [isPinVerified, setIsPinVerified] = useState(false);
```

Comment:
is `isPinVerified` needed when we are waiting for `isDeviceReadyToUseAndAuthorized` anyway?

Reply:
I need to know that handleDeviceChange has been called with !hasDeviceAuthFailed AND isDeviceReadyToUseAndAuthorized. I don't have any better idea how to check that without useState.

Reply:
might this work?

```const handleDeviceChange = useCallback(() => {
        if (hasDeviceAuthFailed) {
            reset();
        }
    }, [hasDeviceAuthFailed, reset]);

    useEffect(() => {
        if (!hasDeviceAuthFailed && isDeviceReadyToUseAndAuthorized) {
            if (navigation.canGoBack()) {
                navigation.goBack();
            }
            reset();
            dispatch(
                removeButtonRequests({
                    device,
                    buttonRequestCode: 'PinMatrixRequestType_Current',
                }),
            );
        }
    }, [device, dispatch, hasDeviceAuthFailed, isDeviceReadyToUseAndAuthorized, navigation, reset]);

Reply:
No, I don't think so. I need to know that DEVICE.CHANGED Connect event has been triggered. 

`isDeviceReadyToUseAndAuthorized` can be true right away if device get locked during app use. It doesn't mean that it's unlocked, only that the device already have state and discovery is running or there are some accounts. 

But maybe that leads to another bug - if I have empty wallet, maybe isDeviceReadyToUseAndAuthorized is falsy all the time 🤔